### DISCUSSION TITLE
Loops in macros

### DISCUSSION THREAD

**Discussion by fmorroni** [KaTeX/KaTeX#3943] [Q&A] [answered]
I want to create a macro `\tabs` that takes an argument and adds `#1` number of `\quad`s. I've tried using `\loop` but it doesn't seem to be supported.
I also didn't find much information regarding custom macros. Is is possible to define one as a function which receives the macro arguments and returns a string representing the macro body? Something like:
```javascript
      macros: {
        '\\tabs': (arg1) => '\\quad'.repeat(arg1),
      }
```
Or is there any way to do what I want?

**Comment by edemaine** [comment, marked as answer]
Yes, it's possible, but I don't think it's documented currently. See `src/macros.js` for a bunch of examples. Dealing with numbers is unfortunately a bit painful, because each character becomes its own token, and the tokens are in reverse order. This should do what you want:
```js
katex.renderToString('\\tabs{10}', {
  macros: {
    '\\tabs': (context) => {
      let [arg] = context.consumeArgs(1);
      arg = arg.reverse().map(token => token.text).join('');
      arg = Number(arg);
      return '\\quad'.repeat(arg)
    }
  }
})
```

**Reply by fmorroni** [reply]
Great that worked, thanks!
